octopus-server/src/test/java/octopus/teamcity/server/generic/CreateReleaseStepTest.java

```diff
@@ -0,0 +1,105 @@
+package octopus.teamcity.server.generic;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import jetbrains.buildServer.serverSide.InvalidProperty;
+import octopus.teamcity.common.createrelease.CreateReleasePropertyNames;
+import org.junit.jupiter.api.Test;
+
+class CreateReleaseStepTest {
+
+  private final CreateReleaseStep step = new CreateReleaseStep();
+
+  @Test
+  public void describePropertiesProducesExpectedOutput() {
+    String message = step.describeParameters(buildPropertiesMap());
+    assertEquals("Project name: Project-1\nPackage version: 1.0.0", message);
+  }
+
+  @Test
+  public void validatePropertiesReturnsEmptyInvalidPropertiesListWithValidProperties() {
+    List<InvalidProperty> invalidProperties = step.validateProperties(buildPropertiesMap());
+    assertNotNull(invalidProperties);
+    assertEquals(0, invalidProperties.size());
+  }
+
+  @Test
+  public void validatePropertiesReturnsSingleInvalidPropertyNullProjectName() {
+    Map<String, String> properties = buildPropertiesMap();
+    properties.put(CreateReleasePropertyNames.PROJECT_NAME, null);
+
+    List<InvalidProperty> invalidProperties = step.validateProperties(properties);
+    assertNotNull(invalidProperties);
+    assertEquals(1, invalidProperties.size());
+    assertEquals(
```

Comment:
nit: its probably worth validating the invalidProperty _field_ is set to CreateReleaePropertyNames.PROJECT_NAME

Reply:
Good idea, have updated tests to check field